### ISSUE TITLE
Local Run not working on 5.0.12

### ISSUE THREAD

**Issue by cadufreeela** [FlutterFlow/flutterflow-issues#4912]
### Can we access your project?

- [X] I give permission for members of the FlutterFlow team to access and test my project for the sole purpose of investigating this issue.

### Current Behavior

FAILURE: Build failed with an exception.

* What went wrong:
Could not open cp_settings generic class cache for settings file '/Users/carloslinares/Library/Application Support/io.flutterflow.prod.mac/freeela/android/settings.gradle' (/Users/carloslinares/.gradle/caches/7.5/scripts/8q20grtjrjwoqx0ui1j7spzt3).
> BUG! exception in phase 'semantic analysis' in source unit '_BuildScript_' Unsupported class file major version 65

### Expected Behavior

It should work on local run

### Steps to Reproduce

1. Open any project on FF 5.0.12
2. Local Run

### Reproducible from Blank

- [X] The steps to reproduce above start from a blank project.

### Bug Report Code (Required)

ITFTksjf35N2ofVY+Krcc8ZsoCAXCUMdXOczitVEew4bCIj4PbQuOc/7P1tWQtSzaHNle2H+uGMKptHFhvDpUcIEPQ+Bbqk4z6hQahPPYz+jaLqUBam/PkElAZ9Yfmqi56uJvCRDNvRec04fwU61f+6/dz2Cf9qOYwx5e6fDbOY=

### Visual documentation

<img width="777" alt="Screenshot 2024-11-16 at 7 01 51 pm" src="https://github.com/user-attachments/assets/b4f3a1e2-092a-461f-bbf0-7865979cc398">
<img width="792" alt="Screenshot 2024-11-16 at 7 08 45 pm" src="https://github.com/user-attachments/assets/98fd14eb-89f4-4d1d-9c5b-1c33ecd49af3">

### Environment

```markdown
- FlutterFlow version: 5.0.12
- Platform:Macos Desktop
- Browser name and version:
- Operating system and version affected: 15.0.1 (24A348)
```

**Comment by stepanic** [comment]
**RESOLVED**

The issue is when you upgrade Android Studio to the latest version and then Kotlin compatibility and JVM target version have problems with the default version.

# Solution 1

Downgrade Android Studio to Koala
https://developer.android.com/studio/releases/past-releases/as-koala-release-notes

# Solution 2

1. Upgrade Gradle to 8.9, recommended with Android Studio with Android Gradle Plugin Upgrade Assistant and sync project to the latest possible version, for me was 8.7.2

2. 
Update `./android/app/build.gradle` in the `android` section add
```
   compileOptions {
        sourceCompatibility = JavaVersion.VERSION_17
        targetCompatibility = JavaVersion.VERSION_17
  }
    
  kotlinOptions {
    jvmTarget = "17"
  }
```

@asmengistu can you please upgrade `./android/app/build.gradle` in the FlutterFlow generator to enable this also permanent in the FlutterFlow Local run.



Solution found here
- https://www.reddit.com/r/FlutterDev/comments/1gqc5lv/for_those_struggling_with_the_new_android_studio/
- https://stackoverflow.com/questions/76069765/upgrading-android-gradle-plugin-to-8-0-0-jvm-target-compatibility-should-be-set

**Comment by paulperez-dev** [comment]
Hi @cadufreeela! Could you please review the solutions that Stepanic shared and let us know if they work for you?

**Comment by cadufreeela** [comment]
@paulperez-dev I'll try tonight, I'm in Australia

**Comment by rzambroni** [comment]
Hi @cadufreela, just following up on this, did the solution work for you?🙂

**Comment by cadufreeela** [comment]
HI,

Yes, it worked with the latest update and Android Koala. You want me to
test with AS Ladybug now and the latest FF update: FlutterFlow v5.0.13
released November 21, 2024 Flutter version is 3.24.2

thanks,
